gateleen-core/src/main/java/org/swisspush/gateleen/core/resource/CopyResourceHandler.java

```diff
@@ -175,23 +175,27 @@ private void createResponse(final HttpServerRequest request, final HttpClientRes
      * @param buffer - the buffer of the request
      * @return a new CopyTask
      */
-    private CopyTask createCopyTask(final HttpServerRequest request, final Buffer buffer) {
+    CopyTask createCopyTask(final HttpServerRequest request, final Buffer buffer) {
         JsonObject task = new JsonObject(buffer.toString());
+        HeaderFunction headerFunction = HeaderFunctions.DO_NOTHING;
 
-        JsonObject staticHeadersJson = task.getJsonObject("staticHeaders");
-        MultiMap staticHeaders = new CaseInsensitiveHeaders();
-        if (staticHeadersJson != null && staticHeadersJson.size() > 0) {
-            for (Map.Entry<String, Object> entry : staticHeadersJson.getMap().entrySet()) {
-                staticHeaders.add(entry.getKey(), entry.getValue().toString());
+        JsonObject staticHeaders = task.getJsonObject("staticHeaders");
+        if (staticHeaders != null) {
+            headerFunction = HeaderFunctions.parseStaticHeadersFromJson(staticHeaders);
+        } else {
+            final JsonArray dynamicHeaders = task.getJsonArray("headers");
+            if (dynamicHeaders != null) {
+                headerFunction = HeaderFunctions.parseFromJson(dynamicHeaders);
             }
         }
 
         MultiMap headers = StatusCodeTranslator.getTranslateFreeHeaders(request.headers());
-        headers = HttpHeaderUtil.removeNonForwardHeaders(headers);
+        headerFunction.apply(headers);
```

Comment:
The HeaderFunction returns an `EvalScope` object which contains error information.
Propose to answer the Http-Request with "400 Bad Request".

See https://github.com/swisspush/gateleen/blob/v1.1.46/gateleen-routing/src/main/java/org/swisspush/gateleen/routing/Forwarder.java#L197-L205 in Gateleen's Router

Reply:
OK, right. Will look into this.